src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,112 @@
+        const input: MessageItem = await ext.ui.showWarningMessage(notConfiguredForDeploy, { modal: true }, DialogResponses.yes, dontShowAgainButton);
+        if (input === dontShowAgainButton) {
+            await updateWorkspaceSetting(configurePreDeployTasksSetting, false, context.workspace.uri.fsPath);
+        } else {
+            // resolves to "."if it is not a subfolder
+            const subfolder: string = path.dirname(path.relative(context.workspace.uri.fsPath, csprojFile));
+            const deploySubpath: string = path.posix.join(subfolder, 'bin', 'Release', 'publish');
+
+            await updateWorkspaceSetting(preDeployTaskSetting, 'publish', context.workspace.uri.fsPath);
+            await updateWorkspaceSetting(constants.configurationSettings.deploySubpath, deploySubpath, workspaceFspath);
+
+            // update the deployContext.deployFsPath with the .NET output path since getDeployFsPath is called prior to this
+            // purposely not using posix since this is happening at runtime
+            context.deployFsPath = path.join(context.workspace.uri.fsPath, deploySubpath);
+
+            const defaultTasks = getTasks(deploySubpath, subfolder);
+            tasks.updateTasks(context.workspace, defaultTasks);
+        }
+
+    }
+
+    async function tryGetCsprojFile(projectPath: string): Promise<string | undefined> {
+        let projectFiles: string[] = await checkFolderForCsproj(projectPath);
+        // it's a common pattern to have the .csproj file in a subfolder so check one level deeper
+        if (projectFiles.length === 0) {
+            for (const folder of fse.readdirSync(projectPath)) {
+                const filePath: string = path.join(projectPath, folder);
+                if (fse.existsSync(filePath) && (await fse.stat(filePath)).isDirectory()) {
+                    projectFiles = projectFiles.concat(await checkFolderForCsproj(filePath));
+                }
+            }
+        }
+
+        context.telemetry.properties.numOfCsprojFiles = projectFiles.length.toString();
+
+        // if multiple csprojs were found, ignore them
+        return projectFiles.length === 1 ? projectFiles[0] : undefined;
+
+        async function checkFolderForCsproj(filePath: string): Promise<string[]> {
+            const files: string[] = fse.readdirSync(filePath);
+            const filePaths: string[] = files.map((f: string) => {
+                return path.join(filePath, f);
+            });
+
+            return filePaths.filter((f: string) => /\.csproj$/i.test(f));
+        }
+    }
+}
+
+function getTasks(deploySubpath: string, subfolder: string): TaskDefinition[] {
+    // tslint:disable-next-line: no-unsafe-any no-invalid-template-strings
+    const cwd: string = path.posix.join('${workspaceFolder}', subfolder);
+    return [
+        {
+            label: 'clean',
+            command: 'dotnet clean',
+            type: 'shell',
+            problemMatcher: '$msCompile',
+            options: {
```

Comment:
You should keep the tasks clean if there is no subfolder and omit `options` entirely